Source/Global/global.cpp

```diff
@@ -70,14 +65,17 @@ HRESULT http_singleton::singleton_access(
 
         if (!s_singleton)
         {
-            assert(!s_useCount);
+            assert(s_useCount == 0);
             return E_HC_NOT_INITIALISED;
         }
 
+        --s_useCount;
         singleton = s_singleton;
-        if (!--s_useCount)
+
+        if (s_useCount == 0)
         {
             s_singleton.reset();
+            return E_HC_INTERNAL_STILLINUSE;
```

Comment:
wrong way round?

Reply:
oh good call

---
In reply to: [377989551](https://github.com/microsoft/libHttpClient/pull/492#discussion_r377989551) [](ancestors = 377989551)